### PR TITLE
Updates to documentation for NEST 3

### PR DESCRIPTION
**Pull request by jessica-mitchell**
Several different changes made to make the documentation look decent on RtD
and update some terminology and formatting.

Note that there is still work to do be done in the part 4 PyNEST tutorials, and the former topology user manual should be checked over.
The PyNEST examples are kind of organized but this could be improved.

### CODE REVIEW

**Comment by jougs** [comment]
Is there a rendered version somewhere already? Can you please point me to it? Thanks!

File: doc/tutorials/pynest_tutorial/part_4_topologically_structured_networks.rst [hunk 1]
```diff
@@ -378,6 +378,9 @@ Section 4.1 *Query functions*.
       'network_size': 20,
       'shape': (5, 4)}
 
+It may also be useful to look at the ``spatial`` property of the
+NodeCollection, which describes the layer properties. Other useful
+functions that may be of help are listed in NTUM Section 4.1.
```
Comment (jougs) [inline doc/tutorials/pynest_tutorial/part_4_topologically_structured_networks.rst]: The term NTUM does not make sense anymore. It meant NEST Topology Users Manual.

File: pynest/examples/spatial/conncomp.py [hunk 2]
```diff
@@ -19,16 +19,17 @@
 # You should have received a copy of the GNU General Public License
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
-'''
-NEST Spatial Example
+"""
+Create two populations of pyramidal cells and two populations of interneurons
+======================================----------------------------------------
```
Comment (jougs) [inline pynest/examples/spatial/conncomp.py]: This headline mark looks weird to me.

**Comment by jessica-mitchell** [comment]
@jougs sorry forgot to add that - here is the rendered version https://nest-test.readthedocs.io/en/nest-3/

**Comment by stinebuu** [comment]
Thanks for this @jessica-mitchell, it is really appreciated!

Could you remove `quickref.html` from `doc/CMakeList.txt` line 71? This is what is making your Travis tests fail.
https://github.com/nest/nest-simulator/blob/00e0d3cc822688e2623d1353db34910e431da957/doc/CMakeLists.txt#L71

@jougs We could also convert *NodeCollection*, *SynapseCollection* and *subnet* to italic after this PR is merged if there are some time issues for @jessica-mitchell?

File: doc/tutorials/pynest_tutorial/part_4_topologically_structured_networks.rst (hunk 1 above)
Comment (jessica-mitchell) [reply doc/tutorials/pynest_tutorial/part_4_topologically_structured_networks.rst]: I think this was added by @stinebuu  as she had started working on the changes in the tutorials - I just wanted to mainly update links in this file.

File: doc/guides/guide_structurally_spatial_networks.rst [hunk 3]
```diff
@@ -1396,7 +1395,7 @@ and SLI documentation.
 +---------------------------------+---------------------------------------------+
 | ``nest.DumpLayerConnections()`` | Write connectivity information to file.     |
 |                                 | This function may be very useful to check   |
-|                                 | that Topology created the correct           |
+|                                 | that NEST created the correct           |
```
Comment (stinebuu) [inline doc/guides/guide_structurally_spatial_networks.rst]:
```suggestion
|                                 | that NEST created the correct               |
```

File: doc/ref_material/nest2_vs_3.rst [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+| list=None, synapse_model=None,              | :green:`nest.NodeCollection` =None,           |
+| synapse_label=None)                         | :green:`nest.NodeCollection` =None,           |
+| *returns* numpy.array                       | synapse_model=None, synapse_label=None)       |
+|                                             | *returns* :darkgreen:`nest.SynapseCollection` |
++---------------------------------------------+-----------------------------------------------+
+| nest.Connect(list, list, conn_spec          | nest.Connect(:green:`nest.NodeCollection`,    |
+| =None, syn_spec=None, model=None)           | :green:`nest.NodeCollection`, conn_spec=      |
+|                                             | None, syn_spec=None,                          |
+|                                             | :green:`return_SynapseCollection`=False)      |
+|                                             | | *In syn_spec the synapse model*               |
```
Comment (stinebuu) [inline doc/ref_material/nest2_vs_3.rst]:
```suggestion
|                                             | | *In syn_spec the synapse model*             |
```

File: doc/ref_material/nest2_vs_3.rst [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+| nest.DataConnect(pre, params=None,          | *Use* nest.Connect(list, list,                |
+| model="static_synapse")                     | conn_spec=None, syn_spec=None,                |
+|                                             | return_SynapseCollection=False)               |
+|                                             | *You should only use nest.Connect with lists* |
+|                                             | *if you have lists with non-unique node IDs.* |
++---------------------------------------------+-----------------------------------------------+
+| nest.CGConnect(list, list, cg,              | nest.CGConnect(:green:`nest.NodeCollection`,  |
+| parameter_map=None, model='static           | :green:`nest.NodeCollection`, cg,             |
+| _synapse')                                  | parameter_map=None,                           |
+|                                             | :green:`synapse_model` ='static_synapse')      |
```
Comment (stinebuu) [inline doc/ref_material/nest2_vs_3.rst]:
```suggestion
|                                             | :green:`synapse_model` ='static_synapse')     |
```

**Comment by jessica-mitchell** [comment]
@jougs @stinebuu Travis error says the apt-get install failed -- it seems a connection failed  - can it be restarted by someone?

File: doc/examples/index.rst [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+   ../auto_examples/Potjans_2014/helpers
+   ../auto_examples/Potjans_2014/network_params
+   ../auto_examples/Potjans_2014/stimulus_params
+   ../auto_examples/Potjans_2014/sim_params
+
+
+
+.. toctree::
+   :maxdepth: 1
+   :caption: MUSIC example
```
[the remark this answers is not in the document]
Comment (jessica-mitchell) [reply doc/examples/index.rst]: I've removed MUSIC from the examples

File: doc/guides/from_nest2_to_nest3.rst [hunk 7]
```diff
@@ -1,448 +1,720 @@
...
 
-    ::
+    >>>  nest.Connect(nodes[:2], nodes[:2])
+    >>>  synColl = nest.GetConnections()
+    >>>  print(synColl)
+         *--------*-------------*
+         | source | 1, 1, 2, 2, |
+         *--------*-------------*
+         | target | 1, 2, 1, 2, |
+         *--------*-------------*
```
[the remark this answers is not in the document]
Comment (jessica-mitchell) [reply doc/guides/from_nest2_to_nest3.rst]: I'd suggest @stinebuu  or @hakonsbm  to maybe help with this

**Pull request merged**